pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,255 @@
+        for k, c in self.charts.items():
+            if k in charts:
+                c.show()
+            else:
+                c.hide()
+        # MOVE LEGEND TO LAYOUT
+        if self.pi.legend is not None:
+            self.pi.legend.setParentItem(self.pi)
+        self.update()
+
+    def clean(self):
+        # CLEAR PLOTS
+        for p in self.charts.values():
+            p.clear()
+        self.update()
+
+    def getPlotItem(self, name=None):
+        if name is None:
+            return self.pi
+        else:
+            return self.charts[name].plotItem
+
+    def setAxisRange(self, axis_name, range=None, **kwargs):
+        if range is None or len(range) == 0:
+            # AUTORANGE
+            range = None
+        elif len(range) == 1:
+            # ZERO TO R
+            range = [min(0, *range), max(0, *range)]
+        elif len(range) == 2:
+            # SET GIVEN RANGE
+            range = [min(*range), max(*range)]
+        else:
+            raise AttributeError("bad range")
+        if range is None:
+            self.enableAxisAutoRange(axis_name)
+        else:
+            self.disableAxisAutoRange(axis_name)
+            axis = self.axis[axis_name]
+            charts = [self.charts[connection] for connection in self.axis_connections[axis_name]]
+            if axis.orientation in ["top", "bottom"]:  # IS X AXIS
+                for chart in charts:
+                    vb = chart.plotItem.getViewBox()
+                    vb.setXRange(*range, **kwargs)
+            elif axis.orientation in ["left", "right"]:  # IS Y AXIS
+                for chart in charts:
+                    vb = chart.plotItem.getViewBox()
+                    vb.setYRange(*range, **kwargs)
+
+    def update(self):
+        for axis_name, connections in self.axis_connections.items():
+            axis = self.axis[axis_name]
+            if axis.autorange:
+                charts = [self.charts[connection] for connection in connections]
+                bounds = []
+                if axis.orientation in ["top", "bottom"]:  # IS X AXIS
+                    for chart in charts:
+                        bounds += chart.dataBounds(ViewBox.XAxis)
+                    bounds = [bound for bound in bounds if bound is not None]
+                    if len(bounds) > 0:
+                        for chart in charts:
+                            vb = chart.plotItem.getViewBox()
+                            vb.setXRange(min(bounds), max(bounds))
+                elif axis.orientation in ["left", "right"]:  # IS Y AXIS
+                    for chart in charts:
+                        bounds += chart.dataBounds(ViewBox.YAxis)
+                    bounds = [bound for bound in bounds if bound is not None]
+                    if len(bounds) > 0:
+                        for chart in charts:
+                            vb = chart.plotItem.getViewBox()
+                            vb.setYRange(min(bounds), max(bounds))
+        super().update()
```

Comment:
some py3 only syntax, which is _totally_ fine now that's we're running python3 😎